File: algorithms/java/src/test/java/leetcode/LC344ReverseString.java
package leetcode;

import org.junit.jupiter.api.Test;


import static org.junit.jupiter.api.Assertions.assertEquals;

/**
 * https://leetcode.com/problems/reverse-string/
 * #easy
 */
public final class LC344ReverseString {
    public final class Solution {
        public String reverseString(final String s) {
            char[] reversed = s.toCharArray();
            for (int i = 0; i < Math.round(reversed.length / 2.0); i++) {
                char exchange = reversed[i];
                reversed[i] = reversed[reversed.length - 1 - i];
                reversed[reversed.length - 1 - i] = exchange;
            }
            return new String(reversed);
        }
    }

    @Test
    public void test_example() throws Exception {
        assertEquals("olleh", new Solution().reverseString("hello"));
    }
}

File: algorithms/code/leetcode/lc344_reverse_string/lc344_reverse_string.py
# https://leetcode.com/problems/reverse-string/

import unittest


class Solution:
    def reverseString(self, s: list[str]) -> None:
        for i in range(round(len(s) / 2)):
            exchange = s[i]
            s[i] = s[len(s) - 1 - i]
            s[len(s) - 1 - i] = exchange


class TestCode(unittest.TestCase):
    def test_example(self) -> None:
        s = list('hello')
        Solution().reverseString(s)
        self.assertListEqual(list('olleh'), s)
